dbms/src/Flash/Planner/plans/PhysicalMockTableScan.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <DataStreams/MockTableScanBlockInputStream.h>
+#include <Flash/Coprocessor/DAGPipeline.h>
+#include <Flash/Coprocessor/GenSchemaAndColumn.h>
+#include <Flash/Coprocessor/MockSourceStream.h>
+#include <Flash/Planner/FinalizeHelper.h>
+#include <Flash/Planner/PhysicalPlanHelper.h>
+#include <Flash/Planner/plans/PhysicalMockTableScan.h>
+#include <Interpreters/Context.h>
+
+namespace DB
+{
+PhysicalMockTableScan::PhysicalMockTableScan(
+    const String & executor_id_,
+    const NamesAndTypes & schema_,
+    const String & req_id,
+    const Block & sample_block_,
+    const BlockInputStreams & mock_streams_)
+    : PhysicalLeaf(executor_id_, PlanType::MockTableScan, schema_, req_id)
+    , sample_block(sample_block_)
+    , mock_streams(mock_streams_)
+{}
+
+PhysicalPlanNodePtr PhysicalMockTableScan::build(
+    Context & context,
+    const String & executor_id,
+    const LoggerPtr & log,
+    const TiDBTableScan & table_scan)
+{
+    NamesAndTypes schema;
+    BlockInputStreams mock_streams;
+    auto & dag_context = *context.getDAGContext();
+    size_t max_streams = dag_context.initialize_concurrency;
+    assert(max_streams > 0);
+    if (dag_context.columnsForTestEmpty() || dag_context.columnsForTest(executor_id).empty())
+    {
+        schema = genNamesAndTypes(table_scan, "mock_table_scan");
+        auto columns_with_type_and_name = getColumnWithTypeAndName(schema);
+        for (size_t i = 0; i < max_streams; ++i)
+            mock_streams.emplace_back(std::make_shared<MockTableScanBlockInputStream>(columns_with_type_and_name, context.getSettingsRef().max_block_size));
```

Comment:
wrap into defaultBuild()?

Reply:
IMO, `defaultBuild` and `buildFromUserInput` are are too trivial....
I wrap a function `mockSchemaAndStreams`